Sources/SwiftSyntax/LRUCache.swift

```diff
@@ -33,12 +32,14 @@ public class LRUCache<Key: Hashable, Value> {
   private unowned var tail: _Node?
 
   public let capacity: Int
+  public private(set) var keysInCache: Set<Key>
```

Comment:
As Alex suggests, I'm not sure this addition is not needed.

But even if we want this API, I don't think this `Set` storage is necessary. This is essentially `table.keys` except `Dictionary.Keys` is not a `Set`. I think something like this should be fine.
```swift
package var keys: some Collection<Key> {
  table.keys
}
```
The caller can create a `Set` from this.